File: VB/WinSolution.Module.Win/MyLookupPropertyEditor.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.XtraEditors.Controls
Imports System.Drawing
Imports DevExpress.Utils
Imports DevExpress.ExpressApp.Win.Editors
Imports DevExpress.ExpressApp.Model

Namespace WinSolution.Module.Win
	Public Class MyLookupPropertyEditor
		Inherits LookupPropertyEditor
		Public Sub New(ByVal objectType As Type, ByVal model As IModelMemberViewItem)
			MyBase.New(objectType, model)
		End Sub
		Protected Overrides Sub SetupRepositoryItem(ByVal item As DevExpress.XtraEditors.Repository.RepositoryItem)
			MyBase.SetupRepositoryItem(item)
			CType(item, RepositoryItemLookupEdit).Init(Me, DisplayFormat, Me.Helper)
			Dim appearance As New AppearanceObject()
			appearance.Font = New Font(FontFamily.GenericSansSerif, 10F, FontStyle.Underline)
			CType(item, RepositoryItemLookupEdit).Buttons.Add(New EditorButton(ButtonPredefines.Glyph, "Edit", -1, True, True, True, HorzAlignment.Default, Nothing, New KeyShortcut(System.Windows.Forms.Keys.Enter), appearance, "Press this button to edit the object"))
			AddHandler (CType(item, RepositoryItemLookupEdit)).ButtonClick, AddressOf MyLookupPropertyEditor_ButtonClick
		End Sub
		Private Sub MyLookupPropertyEditor_ButtonClick(ByVal sender As Object, ByVal e As ButtonPressedEventArgs)
			If (Not e.Button.IsDefaultButton) Then
				Dim objectSpace As IObjectSpace = Helper.Application.CreateObjectSpace()
				Dim editedObject As Object = Nothing
				If BindingHelper IsNot Nothing Then
					editedObject = objectSpace.GetObject(BindingHelper.GetControlValue())
				ElseIf TypeOf sender Is LookupEdit Then
					editedObject = objectSpace.GetObject((TryCast(sender, LookupEdit)).EditValue)
				End If
				If editedObject IsNot Nothing Then
					Dim svp As New ShowViewParameters()
					svp.CreatedView = Helper.Application.CreateDetailView(objectSpace, editedObject, True)
					svp.TargetWindow = TargetWindow.Default
					svp.Context = TemplateContext.View
					svp.CreateAllControllers = True
					Dim svs As New ShowViewSource(Helper.Application.MainWindow, Nothing)
					Helper.Application.ShowViewStrategy.ShowView(svp, svs)
				End If
			End If
		End Sub
	End Class
End Namespace

File: CS/WinSolution.Module.Win/MyLookupPropertyEditor.cs
using System;
using DevExpress.ExpressApp;
using DevExpress.XtraEditors.Controls;
using System.Drawing;
using DevExpress.Utils;
using DevExpress.ExpressApp.Win.Editors;
using DevExpress.ExpressApp.Model;

namespace WinSolution.Module.Win {
    public class MyLookupPropertyEditor : LookupPropertyEditor {
        public MyLookupPropertyEditor(Type objectType, IModelMemberViewItem model) : base(objectType, model) { }
        protected override void SetupRepositoryItem(DevExpress.XtraEditors.Repository.RepositoryItem item) {
            base.SetupRepositoryItem(item);
            ((RepositoryItemLookupEdit)item).Init(this, DisplayFormat, this.Helper);
            AppearanceObject appearance = new AppearanceObject();
            appearance.Font=new Font(FontFamily.GenericSansSerif, 10F, FontStyle.Underline);
            ((RepositoryItemLookupEdit)item).Buttons.Add(new EditorButton(ButtonPredefines.Glyph, "Edit", -1, true, true, true, HorzAlignment.Default, null, new KeyShortcut(System.Windows.Forms.Keys.Enter), appearance, "Press this button to edit the object"));
            ((RepositoryItemLookupEdit)item).ButtonClick += new ButtonPressedEventHandler(MyLookupPropertyEditor_ButtonClick);
        }
        void MyLookupPropertyEditor_ButtonClick(object sender, ButtonPressedEventArgs e) {
            if (!e.Button.IsDefaultButton) {
                IObjectSpace objectSpace = Helper.Application.CreateObjectSpace();
                object editedObject = null;
                if (BindingHelper != null) {
                    editedObject = objectSpace.GetObject(BindingHelper.GetControlValue());
                }
                else if (sender is LookupEdit) {
                    editedObject = objectSpace.GetObject((sender as LookupEdit).EditValue);
                }
                if (editedObject != null) {
                    ShowViewParameters svp = new ShowViewParameters();
                    svp.CreatedView = Helper.Application.CreateDetailView(objectSpace, editedObject, true);
                    svp.TargetWindow = TargetWindow.Default;
                    svp.Context = TemplateContext.View;
                    svp.CreateAllControllers = true;
                    ShowViewSource svs = new ShowViewSource(Helper.Application.MainWindow, null);
                    Helper.Application.ShowViewStrategy.ShowView(svp, svs);
                }
            }
        }
    }
}
